public/app/features/explore/LogsContainer.tsx

```diff
@@ -87,6 +89,30 @@ export class LogsContainer extends PureComponent<LogsContainerProps> {
     return [];
   };
 
+  /**
+   * Get links from the filed of a dataframe that was given to as and in addition check if there is associated
+   * metadata with datasource in which case we will add onClick to open the link in new split window. This assumes
+   * that we just supply datasource name and field value and Explore split window will know how to render that
+   * appropriately. This is for example used for transition from log with traceId to trace datasource to show that
+   * trace.
+   * @param field
+   * @param rowIndex
+   */
+  getFieldLinks = (field: Field, rowIndex: number) => {
+    const data = getLinksFromLogsField(field, rowIndex);
+    return data.map(d => {
+      if (d.link.meta?.datasourceName) {
```

Comment:
I'm not 100 convinced about the meta type on the DataLink model. My initial thinking is that the datasource name could belong to the DataFrame.meta (QueryResultMeta ?). And then you could approach the links from logs field with `getFieldLinksSupplier` maybe and retrieve the meta information from DataFrameView which is available on FieldDisplay? @ryantxu WDYT?

Reply:
> datasource name could belong to the DataFrame.meta (QueryResultMeta ?)

The datasourceName is per field. So each field can specify that this value should be open in particular datasource. Don't see `DataFrame.meta` so not sure if that is hypothetical thing and how that should work. It could theoretically be on field config but then it seems to me like the linking information is in two places and dataLink just seems more natural where to put it. Like instead of having link template, you have datasource + query value template.

> And then you could approach the links from logs field with getFieldLinksSupplier

That would make sense in the long run, right now not sure how this will evolve so thought I would keep it separate. This datasource + query value may not be the final thing. @davkal had an idea of having an internal url scheme something like `grafana://explore/jager?q=1234` that would be routed internally. Also tbh I thought `getFieldLinksSupplier` seemed very much tied to the panels and how we present links there thats why I introduced a more generic `getLinksFromLogsField` next to it to be used in Explore.